sensors:
- name: top_kinect
  type: kinect
  source: /amigo/top_kinect/rgbd
  frame_id: /amigo/top_kinect/openni_rgb_optical_frame
  visualize: 0
  voxel_size: 0.02
  max_range: 2.0
  clearing_padding_fraction: 0.05
  normal_k_search: 15
  association_modules:
  - type: PointNormal
    parameters:
        association_correspondence_distance: 0.3
        position_weight: 2.0
        normal_weight: 1.0
        render_width: 320
        render_max_range: 2.2
        render_voxel_size: 0.02
        normal_k_search: 15
        visualize: 0
  - type: PolygonHeight
    parameters:
        max_range: 2.0
        tolerance: 0.035
        min_cluster_size: 20
        visualize: 0
  segmentation_modules:
  - type: EuclideanClustering
    parameters:
        tolerance: 0.035
        min_cluster_size: 20
        visualize: 0

visualize: 1
world_name: tue.library

plugins:
  - name: localization
    lib: libed_localization_plugin.so
    parameters:
        initial_pose_topic: /amigo/initialpose
        num_particles: 500
        laser_model:
            topic: /amigo/base_laser/scan
            num_beams: 100
            z_hit: 0.95
            sigma_hit: 0.2
            z_short: 0.1
            z_max: 0.05
            z_rand: 0.05
            lambda_short: 0.1
            range_max: 10
        odom_model:
            map_frame: map
            odom_frame: /amigo/odom
            base_link_frame: /amigo/base_link
            alpha1: 0.05   # rot -> trans + strafe
            alpha2: 0.5    # trans -> rot
            alpha3: 0.5    # trans -> trans
            alpha4: 0.5    # rot -> rot
            alpha5: 0.5    # trans -> strafe
  - name: gui_server
    lib: libed_gui_server_plugin.so
    parameters:
        robot_name: amigo
  - name: obstacle_grid_publisher
    lib: libed_occupancy_grid_publisher_plugin.so
    parameters:
        frequency: 10
        resolution: 0.05
        topic: /ed_map
        frame_id: map
        sim_time: 1.0
        object_persistence_time: 10
  - name: tf_publisher
    lib: libed_tf_publisher_plugin.so
    parameters:
        root_frame_id: map

